CACHE-11939: add request cf overriding cache rules compat flags (#20965)
cloudflare/cloudflare-docs: src/content/compatibility-flags/fetch-api-request-cf-overrides-cache-rules.md

---
_build:
  publishResources: false
  render: never
  list: never

name: "Override cache rules cache settings in `request.cf` object for Fetch API"
sort_date: "2023-08-01"
enable_date: "2025-04-02"
enable_flag: "request_cf_overrides_cache_rules"
disable_flag: "no_request_cf_overrides_cache_rules"
---

> This flag changes the behavior of cache when requesting assets via the [Fetch API](/workers/runtime-apis/fetch). Cache settings specified in the `request.cf` object, such as `cacheEverything` and `cacheTtl`, are now given precedence over any [Cache Rules]([/cache/how-to/cache-rules/) set.

Reviewer: ```suggestion
This flag changes the behavior of cache when requesting assets via the [Fetch API](/workers/runtime-apis/fetch). Cache settings specified in the `request.cf` object, such as `cacheEverything` and `cacheTtl`, are now given precedence over any [Cache Rules](/cache/how-to/cache-rules/) set.
```

---
_build:
  publishResources: false
  render: never
  list: never

name: "Override cache rules cache settings in `request.cf` object for Fetch API"
sort_date: "2023-08-01"
enable_date: "2025-04-02"
enable_flag: "request_cf_overrides_cache_rules"
disable_flag: "no_request_cf_overrides_cache_rules"
---

> This flag fixes the behavior of cache when requesting assets via the Fetch API whereby the cache settings specified in the `request.cf` object is given precedence over any Cache Rules set. 

Reviewer 2: ```suggestion
This flag changes the behavior of cache when requesting assets via the Fetch API. The cache settings specified in the `request.cf` object is now given precedence over any Cache Rules set.
```

Reviewer 2: I'd link to Fetch also - https://developers.cloudflare.com/workers/runtime-apis/fetch/
And link to Cache Rules - https://developers.cloudflare.com/cache/how-to/cache-rules/

> 
> This flag will be enabled by default after 2025-04-02.

Reviewer 3: Not required. This appears directly above this content anyway.

```suggestion
```

> With the flag enabled - either by default after 2025-03-26 or by setting the `request_cf_overrides_cache_rules` flag - any cache related settings in a Worker's `request.cf` object such as (`cacheEverything`, `cacheTtl`, etc) will override any cache rules for the target asset in subrequest.

Reviewer 2: ```suggestion
This flag will be enabled by default for any compatibility date after 2025-03-26.
```
